transports/quic/Cargo.toml

```diff
@@ -0,0 +1,46 @@
+[package]
+name = "libp2p-quic"
+version = "0.6.0"
+authors = ["David Craven <david@craven.ch>", "Parity Technologies <admin@parity.io>"]
+edition = "2018"
+description = "libp2p-quic is a noise based quic implementation for rust-libp2p."
+repository = "https://github.com/ipfs-rust/libp2p-quic"
+license = "ISC"
+
+[features]
+noise = ["quinn-noise"]
+tls = ["barebones-x509", "quinn-proto/tls-rustls", "rcgen", "ring", "rustls", "untrusted", "webpki", "yasna"]
+
+[dependencies]
+anyhow = "1.0.41"
+async-global-executor = "2.0.2"
+async-io = "1.6.0"
+barebones-x509 = { version = "0.5.0", optional = true, features = ["webpki", "rustls", "std"] }
+bytes = "1.0.1"
+ed25519-dalek = "1.0.1"
+fnv = "1.0.7"
+futures = "0.3.15"
+if-watch = "0.2.2"
+libp2p-core = { version = "0.30.0", path = "../../core" }
+multihash = { version = "0.14.0", default-features = false }
+parking_lot = "0.11.1"
+quinn-noise = { version = "0.3.0", optional = true }
+quinn-proto = { version = "0.7.3", default-features = false }
+rand_core = "0.5.1"
+rcgen = { version = "0.8.11", optional = true }
+ring = { version = "0.16.20", optional = true }
+rustls = { version = "0.19.1", optional = true, features = ["dangerous_configuration"] }
+thiserror = "1.0.26"
+tracing = "0.1.26"
+udp-socket = "0.1.5"
```

Comment:
This crate looks like a copypaste from https://github.com/quinn-rs/quinn/blob/main/quinn/src/platform/mod.rs. Wouldn't it be more correct to try to pick their code into `quinn_udp` crate in the root of quinn-rs?

Reply:
I guess that would be good, yes. Opened an issue for it https://github.com/quinn-rs/quinn/issues/1178

Reply:
Is this mandatory for merging? it may take a while. the rest should be easy to fix.

I have another project that will take me 1-2 weeks to complete (and since I'm getting payed for it it takes precedence), but I'll address your feedback afterwards when I have more idle time. Thanks!

Reply:
I don't think this is mandatory for merging. I may even take this issue if you are busy.

Reply:
Feel free, that would be cool

Reply:
I opened 2 PRs in the quinn repo extracting UDP code into a separate crate and applying your patches.

Reply:
marking as resolved as @kpp and I decided out of band that we will keep `udp-socket`. reaching consensus with quinn developers was harder than expected.